test/PrintAsObjC/availability.swift

```diff
@@ -148,71 +200,169 @@
+        return -1
+      }
+    }
+  
     @available(macOS, introduced: 10.7, deprecated: 10.9, obsoleted: 10.10)
     @objc var singlePlatCombinedPropertyClass: Availability! {
-	get {
-		return nil
-	    }
+        get {
+            return nil
+        }
     }
     @available(macOS, unavailable, renamed: "anotherPlea", message: "still trapped")
     @objc var platformUnavailableRenameWithMessageProperty: Int {
-	get {
-		return -1
-	    }
+        get {
+            return -1
+        }
     }
 }
 
 // Deliberately a high number that the default deployment target will not reach.
 @available(macOS 999, *)
 extension Availability {
-  @objc func extensionAvailability(_: WholeClassAvailability) {}
-
-
-  @available(macOS, deprecated: 10.10)
-  @objc var propertyDeprecatedInsideExtension: Int {
-	  get {
-		  return 0
-	  }
-  }
+    @objc func extensionAvailability(_: WholeClassAvailability) {}
+    
+    
+    @available(macOS, deprecated: 10.10)
+    @objc var propertyDeprecatedInsideExtension: Int {
+        get {
+            return 0
+        }
+    }
 }
 
 @objc class AvailabilitySub: Availability {
     private override init() { super.init() }
     @available(macOS 10.10, *)
     private override init(x _: Int) { super.init() }
+    @available(*, deprecated, message: "init(deprecatedZ:) was deprecated. Use the new one instead", renamed: "init(z:)")
+    @objc init(deprecatedZ: Int) { super.init() }
+    @objc(initWithNewZ:) init(z: Int) { super.init() }
 }
 
 
 @available(macOS 999, *)
 @objc @objcMembers class WholeClassAvailability {
-  func wholeClassAvailability(_: WholeProtoAvailability) {}
+    func wholeClassAvailability(_: WholeProtoAvailability) {}
 }
 
 @available(macOS 999, *)
 @objc protocol WholeProtoAvailability {
-  func wholeProtoAvailability(_: WholeClassAvailability)
+    func wholeProtoAvailability(_: WholeClassAvailability)
+}
+
+
+@objc(SWTReplacementAvailable) class ReplacementAvailable {
+    @objc(methodReplacingInDeprecatedClassWithFirst:second:) func methodReplacingInDeprecatedClass(first: Int, second: Int) -> Void {}
+}
+
+@available(macOS, deprecated, renamed: "ReplacementAvailable")
+@objc class DeprecatedAvailability {
+    @available(*, deprecated, message: "use method in another class instead", renamed: "ReplacementAvailable.methodInDeprecatedClass(first:second:)")
+    @objc func deprecatedMethodInDeprecatedClass(first: Int, second: Int) -> Void {}
 }
+
+@available(macOS, unavailable, renamed: "ReplacementAvailable")
+@objc class UnavailableAvailability {
+  @available(*, unavailable, message: "use method in another class instead", renamed: "ReplacementAvailable.methodInDeprecatedClass(first:second:)")
+  @objc func unavailableMethodInUnavailableClass(first: Int, second: Int) -> Void {}
+}
+
+@objc(SWTReplacementAvailableProtocol) protocol ReplacementAvailableProtocol {
+  @objc(methodReplacingInDeprecatedClassWithFirst:second:) func methodReplacingInDeprecatedClass(first: Int, second: Int) -> Void
```

Comment:
I'll note that this is the same as the default name; you may want to pick something else.